submitscripts/slurm/sapelo_programs.py

```diff
@@ -29,39 +29,35 @@
 # mpi only
 # set scratch dir to home area but run from submit_dir
 
-molpro_mpi = """module load intel/2023a
-
+molpro_mpi = """
 # to change scratch dir to use local machine scratch
 export SCRATCH_DIR=/scratch/$USER/tmp/$SLURM_JOB_ID
 mkdir -p $SCRATCH_DIR
 export APPTAINER_BIND="$SLURM_SUBMIT_DIR,$SCRATCH_DIR"  # This binds the directory into the container so that output can be written.
 
-mpirun -n $NSLOTS apptainer exec /work/jttlab/containers/molpro_mpipr.sif \
-molpro.exe input.dat --output $SLURM_SUBMIT_DIR/output.dat --nouse-logfile --directory $SCRATCH_DIR
-
+singularity run /work/jttlab/containers/molpro-2024.1.1-mpi-gapr.sif -n $NSLOTS input.dat --output $SLURM_SUBMIT_DIR/output.dat --nouse-logfile --directory $SCRATCH_DIR
 rm $SCRATCH_DIR -r
 
 """
 
 # mpi only
 # copy everything to lscratch to run and set scratch to lscratch
 
-molpro_mpi_lscratch = """module load intel/2023a
-
+molpro_mpi_lscratch = """
 # to change scratch dir to use local machine scratch
 export SCRATCH_DIR=/lscratch/$USER/tmp/$SLURM_JOB_ID
 mkdir -p $SCRATCH_DIR
 export APPTAINER_BIND="$SLURM_SUBMIT_DIR,$SCRATCH_DIR"  # This binds the directory into the container so that output can be written.
 
-mpirun -n $NSLOTS apptainer exec /work/jttlab/containers/molpro_mpipr.sif \
-molpro.exe input.dat --output $SLURM_SUBMIT_DIR/output.dat --nouse-logfile --directory $SCRATCH_DIR
+
+singularity run /work/jttlab/containers/molpro-2024.1.1-mpi-gapr.sif -n $NSLOTS input.dat --output $SLURM_SUBMIT_DIR/output.dat --nouse-logfile --directory $SCRATCH_DIR
```

Comment:
How much has this submit script been tested? I didn't have good luck with the runscript on Sapelo previously - it only ever reliably worked locally for me. Any indications that mpi isn't communicating properly in/outside the container?

Reply:
This is a copy-paste from what I use, and I'm running a job with it now. mpirun is already called from within the container, so it doesn't need to be called outside.

Reply:
For reference, it does have trouble communicating outside of the container, but that behavior is not unique to this submit script or use case.